master with mdbx does not compile

```
error[E0308]: mismatched types
   --> /home/greg/.local/share/cargo/git/checkouts/heed-e741d25df84a0eb8/755ef1e/heed/src/db/polymorph.rs:219:17
    |
219 |                 txn.txn,
    |                 ^^^^^^^ expected *-ptr, found struct `txn::RoTxn`
    |
    = note: expected raw pointer `*mut mdbx_sys::MDBX_txn`
                    found struct `txn::RoTxn<T>`
```

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -30,12 +30,14 @@ jobs:
 
       # Test the LMDB backend
       - name: Run cargo test LMDB
-        uses: actions-rs/cargo@v1
-        with:
-          run: cd heed && cargo test --features 'lmdb serde-json' --no-default-features
+        run: |
+          cd heed
+          cargo clean
+          cargo test --features 'lmdb serde-json' --no-default-features
 
       # Test the MDBX backend
       - name: Run cargo test MDBX
-        uses: actions-rs/cargo@v1
-        with:
-          run: cd heed && cargo test --features 'mdbx serde-json' --no-default-features
+        run: |
+          cd heed
+          cargo clean
+          cargo test --features 'mdbx serde-json' --no-default-features

--- a/heed/src/db/polymorph.rs
+++ b/heed/src/db/polymorph.rs
@@ -216,7 +216,7 @@ impl PolyDatabase {
 
         let result = unsafe {
             mdb_result(ffi::mdbx_dbi_sequence(
-                txn.txn,
+                txn.txn.txn,
                 self.dbi,
                 value.as_mut_ptr(),
                 increment,

--- a/heed/src/txn.rs
+++ b/heed/src/txn.rs
@@ -33,8 +33,10 @@ impl<T> RoTxn<T> {
         result.map_err(Into::into)
     }
 
-    pub fn abort(self) -> Result<()> {
-        abort_txn(self.txn)
+    pub fn abort(mut self) -> Result<()> {
+        abort_txn(self.txn)?;
+        self.txn = ptr::null_mut();
+        Ok(())
     }
 }
 

